apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -156,7 +158,12 @@ defmodule OMG.Watcher.BlockGetter do
         ) :: {:noreply, Core.t()} | {:stop, :normal, Core.t()}
   def handle_info(msg, state)
 
-  def handle_info(:producer, state) do
+  def handle_info(:producer, state), do: do_producer(state)
+  def handle_info({_ref, {:downloaded_block, response}}, state), do: do_downloaded_block(response, state)
+  def handle_info({:DOWN, _ref, :process, _pid, :normal} = _process, state), do: {:noreply, state}
+  def handle_info(:sync, state), do: do_sync(state)
+
+  defp do_producer(state) do
```

Comment:
missed comment again:

> weren't you intending to decorate these new functions?

Reply:
is there a reason only `do_producer` is decorated out of the new functions broken apart?